src/GafferScene/Preview/RendererAlgo.cpp

```diff
@@ -513,18 +532,20 @@ struct LocationOutput
 				updatedAttributes->members()[g_setsAttributeName] = boost::const_pointer_cast<InternedStringVectorData>( setsAttribute );
 			}
 
-			// TODO: clean up
-			CompoundObject::ObjectMap::const_iterator it = attributes->members().find( "linkedLightsSet" ); // TODO: naming needs to be determined
+			CompoundObject::ObjectMap::const_iterator it = attributes->members().find( g_linkedLightsSetAttributeName );
 			if( it != attributes->members().end() )
 			{
-				IECore::StringVectorDataPtr lightNames = new IECore::StringVectorData();
-				std::vector<std::string> &lightNamesWritable = lightNames->writable();
-
 				if( StringDataPtr lightsExpressionData = IECore::runTimeCast<StringData>( it->second ) )
 				{
-					GafferScene::PathMatcher linkedlightsSet = SetExpression::evaluateSetExpression( lightsExpressionData->readable(), scene );
-					linkedlightsSet.paths( lightNamesWritable );
-					updatedAttributes->members()["lightsToBeLinked"] = lightNames; // TODO: naming needs to be determined
+					IECore::StringVectorDataPtr lightNames = new IECore::StringVectorData();
```

Comment:
I should have mentioned that we should probably use the `IECore::LRUCache` to simplify the caching. This would have taken care of the two big problems in this implementation :

1. Multiple threads are reading/writing m_lightLinkExpressions at the same time, which will lead to crashes.
2. Multiple threads may evaluate the same expression because it wasn't in m_lightLinkExpressions when they checked, but was by the time they finished their own evaluation.

Anyway, I've suggested in my main comment that perhaps we should have an EvaluateLightLinks node so we don't need to do anything in the render output layer - let's resolve that discussion before worrying more about the caching.